### PR TITLE
[quant] Don't regard MatchAllNode as node matched

### PR DESCRIPTION
**Pull request by LiamZhuuu**
Summary:
As title, currently in the (add, X, MatchAllNode) pattnern, the node matched with MatchAllNode is regard as part of the pattern instead of the input. As a result, the possible patterns ends with that node will not be matched.

For instance, we have two patterns
1. (nn.ReLU, (torch.add, MatchAllNode, (nn.BatchNorm2d, nn.Conv2d)))
2. (nn.ReLU, (nn.BatchNorm2d, nn.Conv2d))

And we wanna fuse the following model

Conv2d -> BatchNorm2d -> ReLU +
Conv2d -> BatchNorm2d ------ Add -> ReLU

The pattern in the first row cannot be matched becaues the end node ReLU is recorded as MatchAllNode already.

Test Plan: updated the unit test test_fusion_pattern_with_multiple_inputs

Differential Revision: D34873730

### CODE REVIEW

**Comment by jerryzh168** [comment]
looks like it breaks some pre-existing tests, please take a look

File: test/quantization/fx/test_quantize_fx.py [hunk 1]
```diff
@@ -596,6 +596,70 @@ def conv_bn_res_relu_extra_inputs_getter(pattern):
             if node.op == "call_module" and type(named_modules[node.target]) == torch.nn.Conv2d:
                 self.assertTrue(len(node.args) == 2), "Expecting the fused op to have two arguments"
 
+    def test_fusion_pattern_with_matchallnode(self):
+        """This test tests that the node matched by MatchAllNode will be regared as an input
```
Comment (jerryzh168) [inline test/quantization/fx/test_quantize_fx.py:600]: thanks for the comment, can we add a bit more description about how this is tested?

**Review by jerryzh168 — APPROVED** [review]
